_posts/2019-06-28-dropping-support-for-mysql.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: Dropping support for MySQL
+date: 2019-06-28  9:34:21
+author: Lukas Zapletal
+tags:
+- foreman
+---
+
+Foreman is essentially a Ruby on Rails application. A big one. And Ruby on Rails applications usually support multiple relational databases thanks to ActiveRecord stack. That's also the case for Foreman. But it's not that easy.
+
+<!--more-->
+
+Foreman 1.22 currently requires one of the following databases:
+
+* PostgreSQL 9.2 or newer
+* MySQL 5.5 or newer
+
+This is not an extensive list you might think. Supporting two databases must not be that hard. Truth is, it's a challenge.
+
+Those two items on the list are both relational database systems with SQL standard, the problem is that very often behavior is different. Historically, some features were implemented differently, however it's pretty clear that PostgreSQL is now much closer to the RDBM "ideal" for both developers and operations, should I say. There are dozens of things that are just not right in MySQL and we need to identify those problems, find a way to fix them and then carry extra bit of code in our codebase for life. This takes a lot of our time we could invest in features, bugfixing or performance improvements.
```

Comment:
Each DB has it's own merits. We aren't moving to PG because it's "better" but rather because supporting just one has its benefits, and we already have plugins and other services that require it. PG also has weird problems we need to identify and fix, but having less databases to support will reduce the amount of such workarounds and will stop forcing us to always write db-agnostic code.

Reply:
https://about.gitlab.com/2019/06/27/removing-mysql-support/ is a similar blog about the exact same topic. One thing stood out to me:

> As a side note – we don't use MySQL ourselves, and not doing so meant we weren't encountering issues BEFORE our users.

Both commercial vendors (Red Hat and ATIX) both only deploy on PostgreSQL. That means they care about proper support. MySQL always has been best effort by the community. The Foreman project wants to deliver high quality software and that's more than just 'it compiles, ship it'.

That's at least my take on this.

Reply:
I am rewriting the paragraph and dropping all my biased words about MySQL :-)